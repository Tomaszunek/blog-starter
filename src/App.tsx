import * as React from 'react';
import { RouteComponentProps } from 'react-router';
import './App.css';


export namespace App {
  export interface IProps extends RouteComponentProps<void> {
    numbers: any   
  }
}

export default class App extends React.Component<App.IProps> {    
  public render() {
    return (
      <div className="App">
        abc
      </div>
    );
  }
}


