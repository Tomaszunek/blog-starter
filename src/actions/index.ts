export * from './articles';
export * from './products';
export * from './mainpage';
export * from './abilities';
export * from './projects';
export * from './appSystem';

