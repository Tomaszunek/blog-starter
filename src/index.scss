@import 'src/styles/variables.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: $body-background;
}

.container {
  padding: 0px 100px;
  background-color: $primary-background;

  @media only screen and (min-width: $sm) and (max-width: $lg) {
    padding: 0px 80px;
  }
  @media only screen and (min-width: $xs) and (max-width: $sm) {
    padding: 0px 60px;
  }
  @media only screen and (max-width: $xs) {
    padding: 0px 40px;ab
  }   
}