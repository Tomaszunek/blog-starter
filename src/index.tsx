import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Application } from './router';
import { configureStore } from './store';
import { createBrowserHistory } from 'history';
import registerServiceWorker from './registerServiceWorker';
import { Provider } from 'react-redux';
import { Router, Switch } from 'react-router';
import Navigation from 'src/pages/Navigation';
import Footer from 'src/pages/Footer';
import './index.scss';

const store = configureStore();
const history = createBrowserHistory();

ReactDOM.render(
  <Provider store={store}>    
    <Router history={history}>
      <div>
        <Navigation/>
          <Switch>
            <Application />
          </Switch>
        <Footer/>
      </div>
    </Router>   
  </Provider>,  
  document.getElementById('root') as HTMLElement
);
registerServiceWorker();