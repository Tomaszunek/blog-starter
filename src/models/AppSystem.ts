export interface IAppSystem {
    loading: boolean   
} 