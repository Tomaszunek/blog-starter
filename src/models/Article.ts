

export interface IArticle {
    id: number,
    name: string,
    body: string
}
    
  
export namespace ArticleFiltes {
    export enum Filter {
        SHOW_ALL = 'all',
        SHOW_ACTIVE = 'active',
        SHOW_COMPLETED = 'completed'
    }
}