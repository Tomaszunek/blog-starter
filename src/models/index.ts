export * from './ArticleModel';
export * from './ProductModel';
export * from './ContentCategory';
export * from './MainPageContentModel';
export * from './AppSystem';

