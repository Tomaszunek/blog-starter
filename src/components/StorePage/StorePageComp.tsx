import * as React from 'react';
import './storePage.scss'
import FullScreenCont from '../template/fullScreenCont/fullSreenCont';
import { IProductModel } from 'src/models';

import image from '../../logo.svg'
import BreadcrumbsTemplate from '../template/breadcrumbs/breadcrumbs';
import { match } from 'react-router';
import ArticleContainer from '../ArticlePage/articleContainer/articleContainer';
import SearchBar from '../template/searchBar/searchBar';

export default class StorePageComp extends React.Component<IStorePageCompProps, any> {  
  public render() {
    const { products } = this.props;
    return (
      <main>
        <FullScreenCont>        
          <img src={image} alt=""/>          
        </FullScreenCont>
        <SearchBar searchType="product"/> 
        <BreadcrumbsTemplate match={this.props.match}/>
        <ArticleContainer products={products} structure={[4,4]}/>        
      </main>
    );
  }
}

interface IStorePageCompProps {
  products: Array<IProductModel>,
  match: match<void>
}


