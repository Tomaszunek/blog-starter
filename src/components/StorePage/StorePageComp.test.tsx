import * as React from 'react';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import StorePageComp from './StorePageComp'
import { IProductModel } from 'src/models';

describe('ProjectPage', () => {
  describe('Component should render right', () => {
    let app:any;
    const testMatch:any = {
      path: '',
      params: {},
      url: ''
    };
    const products:Array<IProductModel> = [{
        id: 1,
        type: "ability",
        body: "body",
        createdAt: new Date(1545167325 *1000),
        image: "images/placeholder/abc.jpg",
        name: "woda",
        slug: "woda",
        updatedAt: new Date(1545167325 *1000),
        productType: "type"
    }]         
    beforeEach(() => {
      app = mount( 
        <StorePageComp products={products} match={testMatch}/>
      )
    })
    it('render correctly snapshot component', () => {
      const tree = renderer
      .create(<StorePageComp products={products} match={testMatch}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
    })    
    it('should contain project containar', () => {      
      expect(app.find('main').length).toEqual(1)               
    })
    it('should contain elements', () => {
      expect(app.find('.headerImage').length).toEqual(1) 
      expect(app.find('.breadcrumbs').length).toEqual(1)
    })
    it('should have proper props', () => {
      expect(app.instance().props.products).toEqual(products)
      expect(app.instance().props.match).toEqual(testMatch) 
    })   
  });
}); 