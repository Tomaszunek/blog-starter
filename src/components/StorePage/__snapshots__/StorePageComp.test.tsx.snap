// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPage Component should render right render correctly snapshot component 1`] = `
<main>
  <div
    className="fullScreenCont"
  >
    <img
      alt=""
      className="headerImage"
      src={undefined}
    />
  </div>
  <div
    className="searchBar"
  >
    <form
      action=""
    >
      <select
        className="contentType"
        defaultValue={
          Array [
            "article",
          ]
        }
        multiple={true}
        name="findType"
        onChange={[Function]}
      >
        <option
          value="article"
        >
          article
        </option>
        <option
          value="product"
        >
          product
        </option>
      </select>
      <div
        className="categoryCont"
      >
        Categories
      </div>
      <button>
        Find
        product
      </button>
    </form>
  </div>
  <ul
    className="breadcrumbs"
  >
    <li
      className="home"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      className="current"
    >
      <a />
    </li>
  </ul>
  <div
    className="articleConteiner"
  >
    <div
      className="grid grid-four"
    >
      <div
        className="articleTile"
        data-attr={1}
      >
        <a
          href="/product/woda"
        >
          <img
            alt=""
            src="./images/placeholders/images/placeholder/abc.jpg"
          />
          <div
            className="tileCont"
          >
            <div>
              <span>
                woda
              </span>
              <p>
                16.99$
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      className="grid grid-four"
    />
  </div>
</main>
`;
